x-pack/test/functional/apps/canvas/expression.ts

```diff
@@ -44,31 +43,31 @@ export default function canvasExpressionTest({ getService, getPageObjects }: Ftr
         expect(elements).to.have.length(4);
       });
 
+      const codeEditorSubj = 'canvasCodeEditorField';
+
       // find the first workpad element (a markdown element) and click it to select it
       await testSubjects.click('canvasWorkpadPage > canvasWorkpadPageElementContent', 20000);
+      await monacoEditor.waitCodeEditorReady(codeEditorSubj);
 
       // open the expression editor
       await PageObjects.canvas.openExpressionEditor();
+      await monacoEditor.waitCodeEditorReady('canvasExpressionInput');
 
       // select markdown content and clear it
-      const mdBox = await find.byCssSelector('.canvasSidebar__panel .canvasTextArea__code');
-      const oldMd = await mdBox.getVisibleText();
-      await mdBox.clearValueWithKeyboard();
+      const oldMd = await monacoEditor.getCodeEditorValue(0);
+      await monacoEditor.setCodeEditorValue('', 0);
 
       // type the new text
       const newMd = `${oldMd} and this is a test`;
-      await mdBox.type(newMd);
-      await find.clickByCssSelector('.canvasArg--controls .euiButton');
+      await monacoEditor.setCodeEditorValue(newMd, 0);
+      await PageObjects.common.sleep(300);
```

Comment:
This _could_ introduce some flakiness here, how did you come to the 300 ms number here?

Reply:
I set 300ms as a _sorta_ arbitrary number. If I'm understanding correctly, the line below sets a 250ms debounce on updates to the sidebar editor before calling the `onChange` callback. That __might__ be enough time for the expression editor to receive the updates, or I could set this to a larger `sleep`.

https://github.com/elastic/kibana/blob/17e51c6d6fe0c36e87c14c9badc2cb1c94dde33f/x-pack/plugins/canvas/canvas_plugin_src/uis/arguments/editor.tsx#L85

I'm running a new flaky test job with 100 runs here: https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/1607. Do you think that is sufficient or maybe try a larger number?

Reply:
I think 100 runs is definitely sufficient, thank you for doing that! If they all pass, then I'd say that 300ms is long enough to sleep, especially now that I know where the 300ms number comes from.

Reply:
Instead of sleep, how about putting assert statement below in a retry? That way the functional test is not couple to a debounce value

Reply:
> Instead of sleep, how about putting assert statement below in a retry? That way the functional test is not couple to a debounce value

Ooh, good idea!